### PR TITLE
review: feature: CtTypeInformation#isInterface(), #isClass()

### PR DESCRIPTION
**Pull request by pvojtechovsky**

### CODE REVIEW

**Comment by pvojtechovsky** [comment]
The CI hanged?

File: src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java [hunk 1]
```diff
@@ -567,6 +567,20 @@ public boolean removeActualTypeArgument(CtTypeReference<?> actualTypeArgument) {
 	}
 
 	@Override
+	public boolean isClass() {
+		CtType<T> t = getDeclaration();
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java]: getTypeDeclaration?

File: src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java [hunk 2]
```diff
@@ -587,6 +601,16 @@ public boolean isAnnotationType() {
 	}
 
 	@Override
+	public boolean isEnum() {
+		CtType<T> t = getDeclaration();
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java]: getTypeDeclaration?

File: src/main/java/spoon/reflect/declaration/CtTypeInformation.java [hunk 3]
```diff
@@ -82,11 +82,21 @@
 	boolean isLocalType();
 
 	/**
+	 * Returns true if this type is an class. Returns false for others including `enum`.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtTypeInformation.java]: Returns true if this type is a class. Returns false for others (enum, interface, generics, annotation).

File: src/main/java/spoon/reflect/factory/AnnotationFactory.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public AnnotationFactory(Factory factory) {
 	public <T extends Annotation> CtAnnotationType<?> create(CtPackage owner, String simpleName) {
 		CtAnnotationType<T> t = factory.Core().createAnnotationType();
 		t.setSimpleName(simpleName);
-		owner.getTypes().add(t);
+		owner.addType(t);
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/AnnotationFactory.java]: rebase

**Comment by monperrus** [comment]
I like this simple and useful feature

**Comment by surli** [comment]
@monperrus it looks OK to me for merging, do you agree?

**Pull request merged**